### ISSUE TITLE
fresh OSX install on Sierra

### ISSUE THREAD

**Issue by dpastoor** [metrumresearchgroup/mrgsolve#163]
Issue serves to follow mrgsolve and deps required after a completely new OS installed:

After installing command line tools still get the following:

```
make: gfortran-4.8: No such file or directory
make: *** [opk_dlsoda_mrg.o] Error 1
ERROR: compilation failed for package ‘mrgsolve’
```

next step - install brew and pull gfortran/gcc et al. Will keep this as running docs to move to wiki or something as needed

**Comment by dpastoor** [comment]
after installing brew and doing

```
brew install gcc
```

the most recent version of gfortran was pulled automatically (I think at 6+ now) so install still fails

**Comment by dpastoor** [comment]
apparently I have already gone through this next step in #7 lol

**Comment by kylebmetrum** [comment]
https://github.com/metrumresearchgroup/mrgsolve/wiki/compilers-and-mrgsolve#mac-osx-users

**Comment by dpastoor** [comment]
Yep, I'm crashing forward with homebrew, which we don't have actual docset on (and any new issues that sierra might present) - so far looking good with

```
brew tap homebrew/versions
brew install gcc48
```

homebrew versions allows specifying specific versions.

**Comment by dpastoor** [comment]
getting closer, takes forever to build though

![image](https://cloud.githubusercontent.com/assets/3196313/20117996/e074d2ba-a5d0-11e6-9c15-64bc928f56d0.png)

trying mrgsolve again at this point gives:

```
ld: warning: directory not found for option '-L/usr/local/lib/gcc/x86_64-apple-darwin13.0.0/4.8.2'
ld: warning: directory not found for option '-L/usr/local/lib/gcc/x86_64-apple-darwin13.0.0/4.8.2'
ld: library not found for -lgfortran
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [mrgsolve.so] Error 1
ERROR: compilation failed for package ‘mrgsolve’
* removing ‘/Library/Frameworks/R.framework/Versions/3.3/Resources/library/mrgsolve’
```

**Comment by dpastoor** [comment]
woo its alive, ultimately pulled down directly

```
curl -O http://r.research.att.com/libs/gfortran-4.8.2-darwin13.tar.bz2
sudo tar fvxz gfortran-4.8.2-darwin13.tar.bz2 -C /
```

and everything built properly

**Comment by kylebmetrum** [comment]
Are you on Sierra?  And it works with 4.8.2?

**Comment by dpastoor** [comment]
Yep, thus the point of tracking it in the issue!


The only 'hiccup' has been: 

```
gfortran-4.8: warning: couldn’t understand kern.osversion ‘16.1.0
```

on the initial compilation, but that doesn't seem to have impacted anything

**Comment by dpastoor** [comment]
officially in business

![image](https://cloud.githubusercontent.com/assets/3196313/20118577/04ee38e6-a5d3-11e6-9a0a-7bad5bf63356.png)

**Comment by dpastoor** [comment]
any reason to keep this open at this point?

tldr is:

1) install osx command line tools by running `xcode-select --install` in the terminal and accept the agreement

2) Install R from cran (r-project.org)

3) Install homebrew then Install gcc 4.8

```
brew tap homebrew/versions
brew install gcc48
```

This process can take ~25 minutes to bootstrap all of gcc 48

4) Install remaining gfortran deps

```
curl -O http://r.research.att.com/libs/gfortran-4.8.2-darwin13.tar.bz2
sudo tar fvxz gfortran-4.8.2-darwin13.tar.bz2 -C /
```

5) Install mrgsolve

This definitely works. I would wonder if you can get around not doing step 3 and just `brew install gcc` then install gfortran 4.8, which would install very quickly.

**Comment by kylebmetrum** [comment]
Thanks for reporting this; I'd definitely like to log this in a wiki page, but would like to figure out which part should be default recommendation for new users.

When you install osx commandline tools, you get all of the build tools, including `clang`, but no `gfortran`.  If you wanted to just use `clang`, you wouldn't need step 3 (?).  The rest looks like standard `R` + tools install process.  

Will this not work with osx command line tools + `clang`?

**Comment by vjd** [comment]
Finally upgraded to Sierra and went through the first 2 steps of @dpastoor list above. Then I tried installing mrgsolve as @kylebmetrum speculated may work with just `clang` - it did not. 

Then I went on to steps 3 and 4. But step 4 part 2 is failing with the following error -

```
x usr/local/lib/gcc/x86_64-apple-darwin13.0.0/4.8.2/finclude/omp_lib_kinds.mod: Cannot extract through symlink usr/local/lib/gcc
```
I actually get a bunch of such errors and I am trying to resolve the issue by searching around on the specific error. Not much came up but I can confirm that the gcc folder is definitely a symlink. 
 Any suggestions?

**Comment by kylebmetrum** [comment]
`mrgsolve` won't install without the `fortran` compiler ... that issue is separate from the `clang`/`gcc` question.


Something similar here http://stackoverflow.com/questions/41198030/unable-to-install-package-psych-on-macos-using-revolution-r-open-3-2-2 but I'm not sure exactly what the solution was.

**Comment by vjd** [comment]
Finally got it working - thank you for the tip on `Makevars` @kylebmetrum 

_**Note** - I could not install gfortran4.8.2. Got it work by modifying the `Makevars` file as noted in this thread here_ http://stackoverflow.com/questions/37753941/os-x-package-installation-issue-cant-find-gfortran-4-8-to-build-package/37754162#37754162

After failing to extract the 4.8.2 fortran as discussed above in Step4 of @dpastoor comment, I did the following -

1. `brew reinstall gcc --without-multilib` - this installed version 6-1
2.  upgraded R to the latest version - 3.2.2 from CRAN
3. Did the suggested upgrade for `Xquartz` (although this may not be necessary)
4. Followed the steps in the link provided above -
Open the `Makevars` file in `~/.R/` folder. (if you don't have one, just create that folder and file)
Inserted the following lines of code in that file 
```
F77 = gfortran
FC = gfortran
FLIBS = -L/usr/local/gfortran/lib

```
5. FWIW restarted my computer ( I know, I know, this is Mac)
6. installed mrgsolve via github

Tested a couple of models and all are running fine. 

Hope this helps!

**Comment by kylebmetrum** [comment]
Yes ... there is specific recommendation from R:
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#El-Capitan-and-Sierra

**Comment by vjd** [comment]
This is perfect.. a little more reading would have helped!

**Comment by indesertum** [comment]
I get stuck on this step

1. brew reinstall gcc --without-multilib - this installed version 6-1

Error: Permission denied - /usr/local/share/man/man7/fsf-funding-6.7

It has both read and write permissions. Any suggestions on what I could do next?

**Comment by kylebmetrum** [comment]
I have updated in the install instructions to point people to the Sierra configuration as the first option.

**Comment by sagar-m** [comment]
Hi, I noticed that I have x86_64-apple-darwin15.5.0-gfortran but not x86_64-apple-darwin13.4.0-gfortran which is needed to install "ClassInt" package in R 3.4.2 on mac os 10.13.2. Any suggestions?
